crates/burn-train/src/lib.rs

```diff
@@ -25,3 +25,156 @@ pub use learner::*;
+    pub fn dummy_classification_input(
+        classification_type: &ClassificationType,
+    ) -> (ClassificationInput<TestBackend>, Tensor<TestBackend, 2>) {
+        let device = &Default::default();
+        const N_SAMPLES: usize = 200;
+        const N_CLASSES: usize = 4;
+
+        let error_mask = {
+            let mut rng = &mut rand::thread_rng();
+            let change_idx = Tensor::from_floats(
+                (0..N_SAMPLES)
+                    .into_iter()
+                    .choose_multiple(
+                        &mut rng,
+                        (N_SAMPLES as f32 * ERROR_PER_SAMPLE_RATE) as usize,
+                    )
+                    .as_slice(),
+                device,
+            );
+            let values = change_idx.ones_like();
+            let mask =
+                Tensor::zeros(Shape::new([N_SAMPLES]), device).scatter(0, change_idx.int(), values);
+            mask.unsqueeze_dim(1).bool()
+        };
+
+        let (targets, changed_targets) = match classification_type {
+            ClassificationType::Binary => {
+                let targets = Tensor::<TestBackend, 2>::random(
+                    Shape::new([N_SAMPLES, 1]),
+                    Distribution::Bernoulli(TRUE_POSITIVE_RATE),
+                    device,
+                )
+                .bool();
+
+                let changed_targets = targets.clone().not_equal(error_mask);
+                (targets, changed_targets.float())
+            }
+            ClassificationType::Multiclass => {
+                let mut classes_changes =
+                    Tensor::<TestBackend, 2>::random([N_SAMPLES, 2], Distribution::Default, device)
+                        .mul_scalar(4)
+                        .int()
+                        .float()
+                        .chunk(2, 1);
+                let (classes, changes) = (classes_changes.remove(0), classes_changes.remove(0));
+                let changed_classes = (classes.clone()
+                    + changes.clamp(1, 2) * error_mask.clone().float())
+                    % N_CLASSES as f32;
+                (
+                    one_hot_encode(classes, N_CLASSES).bool(),
+                    one_hot_encode(changed_classes, N_CLASSES),
+                )
+            }
+            ClassificationType::Multilabel => {
+                let targets = Tensor::<TestBackend, 2>::random(
+                    [N_SAMPLES, N_CLASSES - 1],
+                    Distribution::Default,
+                    device,
+                )
+                .greater_elem(THRESHOLD);
+
+                (targets.clone(), (targets.float() + error_mask.float()) % 2)
+            }
+        };
+        let predictions = changed_targets
+            .random_like(Distribution::Uniform(0.0, THRESHOLD - 0.1))
+            .sub(changed_targets.clone())
+            .abs();
+
+        (
+            ClassificationInput::new(predictions, targets.clone()),
+            targets.float().sub(changed_targets),
+        )
+    }
```

Comment:
Reusing a mock input for multiple tests is a great idea, but 1) it should be split for different use cases tested (e.g., binary, multiclass and multilabel) and 2) it doesn't need to be so complicated to generated a mock input 😅 

I think using a dummy input with a smaller number of samples and hardcoded predictions / target errors makes it easier for everyone. That way, you also don't have to write tests for the functions you added to reuse come code in your tests 😁 (as with the added `test_predictions_targets_match` and `test_error_rate`).

Reply:
Yeah that's fair enough, I'll reduce this complicated mess to some static examples for binary, multi class and multi label examples. Thank you

Reply:
You can use the function to generate an input if you want since it has been validated. Just copy the values of the tensor so it's fixed and then the function can go 😄 (you can probably reduce the number of samples)

Reply:
all done